### PR TITLE
This patch fixes PyJWT support when pyjwt >= 0.4.0

### PR DESCRIPTION
**Pull request by josephtate**
Newer versions of pyjwt subsume a bunch of the PEM decoding stuff.

This fixes tests with all latest deps.

I also reformatted the PEM key strings so that they could be copied/pasted easier. PEP-8 doesn't forbid multi-line strings.

Note that the Private Keys were retrieved by running RSA.importKey(private_pem).publickey.exportKey().

### CODE REVIEW

**Comment by josephtate** [comment]
The CI build failed because tox.ini isn't being respected. PyJWT now requires "cryptography" vs. PyCrypto.

**Comment by lepture** [comment]
ping @ib-lundgren

**Comment by tilgovi** [comment]
LGTM

**Comment by wiliamsouza** [comment]
:+1:  LGTM

**Comment by jpadilla** [comment]
Yesterday we released PyJWT [v1.0.0](https://github.com/jpadilla/pyjwt/releases/tag/1.0.0) which fixes some reported security vulnerabilities. Although this project doesn't seem to be affected, it would still be good idea to require >= 1.0.0 instead.

**Comment by josephtate** [comment]
@jpadilla Thanks for the heads up

**Comment by ib-lundgren** [comment]
Sorry for not having looked at this earlier. I just merged #310 which changed the travis/tox setup a bit, can you please sync and set requirements for pyjwt as needed? 

I see cryptography being installed but never imported / explicitly used. I'm guessing pyjwt does not install this implicitly and we have do it?

**Comment by tilgovi** [comment]
Only if we need to test the asymmetric ciphers.

**Comment by jpadilla** [comment]
PyJWT now allows you to either use pycrypto or cryptography(recommended) for asymmetric ciphers, which this project does use (RS256) to be exact. None of them are installed by default.

**Comment by josephtate** [comment]
I'm working on re-merging. I'm also realizing that pycrypto could possibly be removed, and that there are docs that need to be fixed up.

**Comment by ib-lundgren** [comment]
OAuth1 uses pycrypto as well. I begun a CL to change to cryptography / use either but cryptography was not quite ready at the time. Since then I've not been able to finish it up. Feel free to resume it #226 :)

**Comment by josephtate** [comment]
@jpadilla @ib-lundgren Can I get a review of this? I don't know that I like the way that I'm handling oauth1 signatures sans pycrypto, but cryptography doesn't seem to have its low level signing routines polished very well.

I'll work on updating docs.

File: oauthlib/common.py [hunk 1]
```diff
@@ -243,23 +240,16 @@ def generate_signed_token(private_pem, request):
...
     import jwt
 
-    public_key = RSA.importKey(private_pem).publickey()
-
-    try:
-        # return jwt.verify_jwt(token.encode(), public_key)
-        return jwt.decode(token, public_key)
-    except:
-        raise Exception
+    return jwt.decode(token, public_pem)
```
Comment (jpadilla) [inline oauthlib/common.py]: I'd suggest specifying the allowed algorithms for decoding, in this case it'd be something like:

```python
return jwt.decode(token, public_pem, algorithms=['RS256'])
```

**Comment by josephtate** [comment]
I think I'm all done with this @ib-lundgren.

**Comment by lepture** [comment]
@josephtate maybe you should rebase your commits, make it clear.

**Comment by josephtate** [comment]
Welp, That was fun. I rebased, but that broke, so I re-patched master and force pushed the branch.

Summary:
This current version happens to replace #226, and accidentally includes #336 (which is minor).

**Comment by ib-lundgren** [comment]
Amazing :) Well done and thanks for the help reviewing this all!

**Pull request merged**